internal/api/v1/queryparams/query_params_builder.go

```diff
@@ -0,0 +1,109 @@
+package queryparams
+
+import (
+	"errors"
+	"net/url"
+
+	goclienterr "github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/bitcoin-sv/spv-wallet/models/filter"
+)
+
+type Parser interface {
+	Parse() (url.Values, error)
+}
+
+type Builder struct {
+	parsers []Parser
+}
+
+func (b *Builder) Build() (*URLValues, error) {
+	params := NewURLValues()
+	for _, p := range b.parsers {
+		values, err := p.Parse()
+		if err != nil {
+			return nil, errors.Join(err, goclienterr.ErrQueryParamsBuildFailed)
+		}
+
+		if len(values) > 0 {
+			params.Append(values)
+		}
+	}
+
+	return params, nil
+}
+
+func NewBuilder[F queries.QueryFilters](query *queries.Query[F]) (*Builder, error) {
+	var parsers []Parser
+	if query.Metadata != nil {
+		parsers = append(parsers, &MetadataParser{Metadata: query.Metadata, MaxDepth: DefaultMaxDepth})
+	}
+
+	if query.PageFilter != (filter.Page{}) {
+		parsers = append(parsers, &FilterParser{Filter: query.PageFilter})
+	}
+
+	parsers = append(parsers, initNonAdminQueryFilters(query)...)
+	parsers = append(parsers, initAdminQueryFilters(query)...)
+	return &Builder{parsers: parsers}, nil
+}
+
+func initNonAdminQueryFilters[F queries.QueryFilters](query *queries.Query[F]) []Parser {
+	switch f := any(query.Filter).(type) {
+	case filter.XpubFilter:
+		return []Parser{&FilterParser{Filter: f.ModelFilter}, &FilterParser{Filter: f}}
+
+	case filter.AccessKeyFilter:
+		return []Parser{&FilterParser{Filter: f.ModelFilter}, &FilterParser{Filter: f}}
+
+	case filter.ContactFilter:
+		return []Parser{&FilterParser{Filter: f.ModelFilter}, &FilterParser{Filter: f}}
+
+	case filter.PaymailFilter:
+		return []Parser{&FilterParser{Filter: f.ModelFilter}, &FilterParser{Filter: f}}
+
+	case filter.TransactionFilter:
+		return []Parser{&FilterParser{Filter: f.ModelFilter}, &FilterParser{Filter: f}}
+
+	case filter.UtxoFilter:
+		return []Parser{&FilterParser{Filter: f.ModelFilter}, &FilterParser{Filter: f}}
+	}
+
+	return nil
+}
```

Comment:
I see its a little repetitive, redundant in the cases. What about something like this? Ofcource maybe different approach/naming but you get the gist :D 

```suggestion
func createParsers(f any) []Parser {
    return []Parser{&FilterParser{Filter: f.ModelFilter}, &FilterParser{Filter: f}}
}

switch f := any(query.Filter).(type) {
case filter.XpubFilter, filter.AccessKeyFilter, filter.ContactFilter, filter.PaymailFilter, filter.TransactionFilter, filter.UtxoFilter:
    return createParsers(f)
}
```

Reply:
Try to find a way to skip the switch case and make it more generic/general, in a way that when we add new filter type we will not have to add anything here.